src/pandas_profiling/model/summarizer.py

```diff
@@ -0,0 +1,178 @@
+from functools import reduce
+from typing import Type
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+from visions import VisionsBaseType
+
+from pandas_profiling.model.summary_algorithms import (
+    describe_categorical_1d,
+    describe_counts,
+    describe_date_1d,
+    describe_file_1d,
+    describe_generic,
+    describe_image_1d,
+    describe_numeric_1d,
+    describe_path_1d,
+    describe_supported,
+    describe_url_1d,
+)
+from pandas_profiling.model.typeset import (
+    URL,
+    Boolean,
+    Categorical,
+    DateTime,
+    File,
+    Image,
+    Numeric,
+    Path,
+    ProfilingTypeSet,
+    Unsupported,
+)
+
+
+def compose(functions):
+    """
+    Compose a sequence of functions
+    :param functions: sequence of functions
+    :return: combined functions, e.g. [f(x), g(x)] -> g(f(x))
+    """
+    return reduce(lambda f, g: lambda *x: f(*g(*x)), reversed(functions), lambda *x: x)

```

Comment:
It seems counterintuitive to compose functions reverse from the order provided by the user. Is there a semantic reason to compose them this way in particular?

Reply:
Ahh because of the line graph construction?